symbols/arm7.yml

```diff
@@ -3,55 +3,101 @@ arm7:
     - EU
     - NA
     - JP
+    - EU-WRAM
+    - NA-WRAM
+    - JP-WRAM
+    - EU-RAM
+    - NA-RAM
+    - JP-RAM
   address:
     EU: 0x2380000
     NA: 0x2380000
     JP: 0x2380000
+    EU-WRAM: 0x37F8000
+    NA-WRAM: 0x37F8000
+    JP-WRAM: 0x37F8000
+    EU-RAM: 0x27E0000
+    NA-RAM: 0x27E0000
+    JP-RAM: 0x27E0000
   length:
-    EU: 0x27080
+    EU: 0x27080    #0x1E8 long ENTRY block
     NA: 0x27080
     JP: 0x27080
+    EU-WRAM: 0xF608
+    NA-WRAM: 0xF608
+    JP-WRAM: 0xF608
+    EU-RAM: 0x17878
+    NA-RAM: 0x17878
+    JP-RAM: 0x17878
   description: |-
     The ARM7 binary.
     
     This is the secondary binary that gets loaded when the game is launched.
     
-    Speaking generally, this is the program run by the Nintendo DS's secondary ARM7TDMI CPU, which handles the audio engine, the touch screen, Wi-Fi functions, cryptography, and more.
+    Speaking generally, this is the program run by the Nintendo DS's secondary ARM7TDMI CPU, which handles the audio I/O, the touch screen, Wi-Fi functions, cryptography, and more.
   functions:
     - name: EntryArm7
       address:
         EU: 0x2380000
         NA: 0x2380000
         JP: 0x2380000
       description: |-
-        The entrypoint for the ARM7 CPU. This is like the "main" function for the ARM7 subsystem.
```

Comment:
Can we mention how this function calls `MainArm7` after it sets everything up?

Reply:
> Can we mention how this function calls `MainArm7` after it sets everything up?

It's done the same way as how `NitroMain` is called for arm9 though with a constant being read containing the address to `MainArm7` and then `bx`, should I update the arm9 too?
(Side note, I didn't name it the same since I can't guarantee this is the same function from the NitroSDK. Another thing is that it would also create a name conflict for two functions with completely different contents.)

Reply:
Actually is `NitroMain` named as such because it matches an official NitroSDK leak? If not, since the call pattern is so similar we could also call this the NitroMain for arm7, afaik Nintendo didn't let anyone add code into arm7 so this is definitely Nintendo's main function at least.

Reply:
> It's done the same way as how NitroMain is called for arm9 though with a constant being read containing the address to MainArm7 and then bx, should I update the arm9 too?

Hmm might as well, sounds helpful.

> (Side note, I didn't name it the same since I can't guarantee this is the same function from the NitroSDK. Another thing is that it would also create a name conflict for two functions with completely different contents.)

Yeah agree with the reasoning here.

Reply:
I believe the NitroSDK name for `MainArm7` is `NitroSpMain`, if you want to follow the ARM9 `NitroMain`.

As for `EntryArm7`, that is required to be called `_start` in the actual codebase, just like the ARM9 entrypoint.

Reply:
> I believe the NitroSDK name for `MainArm7` is `NitroSpMain`, if you want to follow the ARM9 `NitroMain`.
> 
> As for `EntryArm7`, that is required to be called `_start` in the actual codebase, just like the ARM9 entrypoint.

I see, I will change `MainArm7` then. For `_start` though, the signature will match between arm7 and arm9 but the two entry functions contain completely different code, is it ok that we use the same name twice here?

Reply:
> Can we mention how this function calls MainArm7 after it sets everything up?
Added, and also on arm9 entry

Reply:
I would just leave it as `EntryArm7`. We can change it later.